test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -0,0 +1,703 @@
+
+        newArray = newArray.filter(column => column.Header === "Benchmark Name" || checkedValues.includes(column.id));
+        this.setState({columns: newArray});
+    }
+    // Update table contents based on which benchmarks are ticked in the benchmark filter
+    onBenchmarkChange = value => {
+        this.setState({benchmarkFilter: value});
+        this.benchmarkFilter(value);
+    }
+    // Date picker format needs to be converted to match the one available in the database entry
+    dateTransform(date, type) {
+        const dateSplit = date.split('/');
+        const jdkDate = dateSplit[2] + ((dateSplit[0].length < 2) ? "0" + dateSplit[0] : dateSplit[0]) + ((dateSplit[1].length < 2) ? "0" + dateSplit[1] : dateSplit[1]);
+
+        if (type === 'test') {
+            this.setState({testJdkDate: jdkDate});
+        } else {
+            this.setState({baselineJdkDate: jdkDate});
+        }
+    }
+    // Based on data returned from database, create relevant columns for platforms
+    generateColumns(platforms) {
+        const newArray = [];
+        let column = {
+        Header: 'Benchmark Name',
+        accessor: 'benchmarkName',
+        Cell: props => <span>{props.value.split(",")[0]} <br/> {props.value.split(",")[1]} <br/> {props.value.split(",")[2]}</span>
+        };
+
+        newArray.push(column);
+
+        for (let i = 0; i < platforms.length; i++) {
+            let platform = platforms[i];
+    	
+            column = {};
+            column.id = platform;
+            column.Header = platform.toUpperCase();
+            column.accessor = d => d.platforms[platform];
+            // Each cell needs to display the comparison by default, on hover displays further details
+            column.Cell = props => <Tooltip title={<div >
+                Raw Test Score: {this.handleProp(props.value, 'testScore')} <br/> 
+                Test CI: {this.handleProp(props.value, 'testCI')}  <br/>
+                Test JDK: {this.handleProp(props.value, 'testJdk')}	<br/> 
+                Test Sdk Resource: {this.handleProp(props.value, 'testSdkResource')}	<br/> 
+                Raw Baseline Score: {this.handleProp(props.value, 'baselineScore')} <br/> 
+                Basline CI: {this.handleProp(props.value, 'baselineCI')}  <br/>
+                Baseline JDK: {this.handleProp(props.value, 'baselineJdk')} <br/>
+                Baseline Sdk Resource: {this.handleProp(props.value, 'baselineSdkResource')}
+                </div> }>
+                <span onClick={() => this.handleLink(this.handleProp(props.value, 'buildUrl'))}> {this.handleProp(props.value, 'compare')} </span></Tooltip>;
+            column.getProps = (state, rowInfo) => (this.handleRegression(this.handleProp(rowInfo.row[platform], 'compare')));
+            newArray.push(column);
+        }
+        this.setState({columns:newArray, originalColumns: newArray});
+    }
+    // Set cell color based on comparison value
+    handleRegression (val) {
```

Comment:
We need to handle 2 cases, which right now we are just handling as one:
1) One of the builds is missing data
2) Relative comparison value is way more than 100 (i.e. 200x or 1000x).